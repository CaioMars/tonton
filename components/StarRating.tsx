import React from "react";
import { FontAwesome } from "@expo/vector-icons";
import { View, Text, StyleSheet } from "react-native";

export default function StarRating({
  rating,
  count,
  size,
}: {
  rating: number;
  count: number;
  size?: number;
}) {
  const whole = Math.floor(rating);
  const isHalf = rating - whole > 0.5 ? true : false;

  const color = "#FFE000";
  const defaultSize = 20;

  function buildStars() {
    const stars = [];
    for (let i = 0; i < whole; i++) {
      stars.push(
        <FontAwesome
          key={i}
          name="star"
          color={color}
          size={size || defaultSize}
        />
      );
    }

    if (isHalf)
      stars.push(
        <FontAwesome
          key={stars.length}
          name="star-half"
          color={color}
          style={{ marginRight: 4 }}
          size={size || defaultSize}
        />
      );

    if (stars.length < 5) {
      let j = 1;
      for (let i = 5 - stars.length; i > 0; i--) {
        stars.push(
          <FontAwesome
            key={stars.length + j}
            name="star-o"
            color={color}
            size={size || defaultSize}
          />
        );
        j++;
      }
    }

    return stars;
  }

  return (
    <View style={styles.container}>
      {buildStars()}
      <Text style={styles.textRating}>{rating}</Text>
      <Text style={styles.textCount}>({count})</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 10,
  },
  textRating: {
    fontWeight: "bold",
    marginHorizontal: 5,
  },
  textCount: {},
});
