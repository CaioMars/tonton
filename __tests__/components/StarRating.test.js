import * as React from "react";
import renderer from "react-test-renderer";
import * as Font from "expo-font";
import { FontAwesome } from "@expo/vector-icons";

import StarRating from "../../components/StarRating";

describe("<StarRating />", () => {
  it(`renders correctly`, async () => {
    await Font.loadAsync({
      ...FontAwesome.font,
    });
    const tree = await renderer
      .create(<StarRating rating={3.5} count={100} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
