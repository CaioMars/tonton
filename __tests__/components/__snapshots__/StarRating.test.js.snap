// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFE000",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFE000",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFE000",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFE000",
          "fontSize": 20,
        },
        Object {
          "marginRight": 4,
        },
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFE000",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "fontWeight": "bold",
        "marginHorizontal": 5,
      }
    }
  >
    3.5
  </Text>
  <Text
    style={Object {}}
  >
    (
    100
    )
  </Text>
</View>
`;
